METALIUM_GUIDE.md

```diff
@@ -89,71 +89,124 @@ The vector addition operation demonstrates the three-kernel architecture clearly
+    const uint32_t a_addr = get_arg_val<uint32_t>(0);
+    const uint32_t b_addr = get_arg_val<uint32_t>(1);
+    const uint32_t n_tiles = get_arg_val<uint32_t>(2);
 
-    const uint32_t tile_size_bytes = get_tile_size(cb_in0);
+    constexpr auto cb_in0 = tt::CBIndex::c_0;
+    constexpr auto cb_in1 = tt::CBIndex::c_1;
 
-    const InterleavedAddrGenFast<true> a = {a_addr, tile_size_bytes, DataFormat::Float16_b};
-    const InterleavedAddrGenFast<true> b = {b_addr, tile_size_bytes, DataFormat::Float16_b};
+    const uint32_t tile_size_bytes = get_tile_size(cb_in0);
 
-    // Read from DRAM into circular buffers
+    const InterleavedAddrGenFast<true> a = {
+        .bank_base_address = a_addr,
+        .page_size = tile_size_bytes,
+        .data_format = DataFormat::Float16_b
+    };
+    const InterleavedAddrGenFast<true> b = {
+        .bank_base_address = b_addr,
+        .page_size = tile_size_bytes,
+        .data_format = DataFormat::Float16_b
+    };
+
+    // Read inputs from DRAM into circular buffers
     for (uint32_t i = 0; i < n_tiles; i++) {
-        cb_reserve_back(tt::c_1, 1);
-        cb_reserve_back(tt::c_0, 1);
-        noc_async_read_tile(get_write_ptr(tt::c_0), a, cb_in0_addr);
-        noc_async_read_tile(get_write_ptr(tt::c_1), b, cb_in1_addr);
+        cb_reserve_back(cb_in0, 1);
+        cb_reserve_back(cb_in1, 1);
+
+        const uint32_t cb_in0_addr = get_write_ptr(cb_in0);
+        const uint32_t cb_in1_addr = get_write_ptr(cb_in1);
+        noc_async_read_tile(i, a, cb_in0_addr);
+        noc_async_read_tile(i, b, cb_in1_addr);
 
         noc_async_read_barrier();  // Wait until tile reads are done
-        cb_push_back(tt::c_0, 1);
-        cb_push_back(tt::c_1, 1);
+
+        cb_push_back(cb_in0, 1);
+        cb_push_back(cb_in1, 1);
     }
 }
 ```
 
-Beyond synchronizing with the reader and writer. The compute kernel needs its own synchronization as internally it is 3 cores running cooperatively. After synchronizing both internally and with the reader and writer, the compute kernel can proceed with actual computation.
+For the compute kernel, the same source code is used to create kernels for all three Unpack/Math/Pack baby RISC-V cores. The Metalium compute API is implemented with code sections that are each enabled only for a specific type of core. In the example below, the `add_tiles` call has Unpack core code that copies the input tiles into the FPU's registers, as well as Math core code that performs the addition operation on the FPU and stores the result into the output register. On the other hand, the `cb_reserve_back` call only contains code that will execute on the Pack core. During execution, the runtime will automatically compile the compute kernel code three times, each time for a different core. The generated binaries will only contain code dedicated for a single type of core and will then be submitted to the device for concurrent execution.
```

Comment:
> For the compute kernel, the same source code is used to create kernels for all three Unpack/Math/Pack baby RISC-V cores.....

I propose we make it the following to reduce confusion.

For the compute kernel, the same source code is used to create **binaries** for all three...